backend/fms_core/template_importer/row_handlers/sample_qc/sample.py

```diff
@@ -0,0 +1,80 @@
+            barcode=sample['container']['barcode'],
+            coordinates=sample['coordinates'],
+        )
+
+        if sample_obj:
+            # Update sample with sample_information
+            new_volume = None
+            if all([sample_information['initial_volume'], sample_information['measured_volume'],
+                    process_measurement['volume_used']]):
+                delta_volume = sample_information['measured_volume'] - sample_information['initial_volume']
+                new_volume = sample_obj.volume + delta_volume - process_measurement['volume_used']
+            else:
+                self.errors['volume'] = 'Initial Volume, Measured Volume and Volume Used are required.'
+
+            if not sample_information['concentration']:
+                self.errors['concentration'] = 'Concentration is required'
+
+            _, self.errors['sample_update'], self.warnings['sample_update'] = \
+                update_sample(sample_to_update=sample_obj,
+                              volume=new_volume,
+                              concentration=sample_information['concentration'])
+
+            # Update derived_sample's flags with sample information
+            _, self.errors['flags'], self.warnings['flags'] = update_qc_flags(sample_obj,
+                                                                              sample_information['quantity_flag'],
+                                                                              sample_information['quality_flag'])
+
+            process_measurement_obj, self.errors['process_measurement'], self.warnings['process_measurement'] = \
+                create_process_measurement(
+                    process=process_measurement['process'],
+                    source_sample=sample_obj,
+                    execution_date=process_measurement['execution_date'],
+                    volume_used=process_measurement['volume_used'],
+                    comment=process_measurement['comment'],
+                )
+
+            if process_measurement_obj:
+                # Validate instruments according to platform
+                for instrument in INSTRUMENT_PROPERTIES:
+                    try:
+                        type = process_measurement_properties[instrument]['value']
+                        it = InstrumentType.objects.get(type=type)
+                        # Validate platform and type
+                        if it.platform.name != TYPES_BY_PLATFORM[it.type]:
+                            self.errors['instrument type'] = f'Invalid type: {it.platform} for instrument: {it.type}.'
+                    except Exception as e:
+                        self.errors['instrument'] = f'Invalid instrument {type}.'
+
+                # Validate required RIN for RNA
+                if sample_obj.derived_sample_not_pool.sample_kind.name == 'RNA' and process_measurement_properties['RIN']['value'] is None:
+                    self.errors['RIN'] = 'It has to be specified for RNA.'
```

Comment:
The message will be very mysterious ... "It" should be more clearly stated as RIN.